Index: index.php
===================================================================
--- index.php	(revision 34785)
+++ index.php	(working copy)
@@ -13,5 +13,15 @@
  */
 define('WP_USE_THEMES', true);
 
+// Enable XHProf profile collection:
+xhprof_sample_enable();
+register_shutdown_function(function () {
+  $url_parts = implode("_", arg());
+  $filename = '/var/platform/xhprof/' . $url_parts . '.' . uniqid() . '.sample_xhprof';
+  file_put_contents($filename, serialize(xhprof_sample_disable()));
+  chmod($filename, 0777);
+});
+
+
 /** Loads the WordPress Environment and Template */
 require( dirname( __FILE__ ) . '/wp-blog-header.php' );
